service/src/main.rs

```diff
@@ -619,52 +622,12 @@ fn subscribe_to_parentchain_new_headers<E: EnclaveBase + Sidechain>(
 			new_header.number
 		);
 
-		last_synced_header = sync_parentchain(enclave_api, api, last_synced_header);
+		let parentchain_block_syncer =
+			ParentchainBlockSyncer::new(api.clone(), enclave_api.clone());
```

Comment:
Constructing a `ParentchainBlockSyncer` should not happen inside the loop. You construct it once outside and then use it inside the loop.